usecases/no2-monitoring/index.md

```diff
@@ -25,44 +59,325 @@ extent = {
         [7.737228350528245,51.86687168604513]
     ]]
 }
-moving_average_window = 31
-method = "max" # mean or max or min
+datacube = connection.load_collection("TERRASCOPE_S5P_L3_NO2_TD_V1", spatial_extent = extent, temporal_extent = [f"{year}-01-01", f"{year}-12-31"])
+```
+
+</template>
+<template v-slot:js>
+
+```js
+let builder = await connection.buildProcess();
+
+let year = 2019;
+let extent = { // Münster
+    "type": "Polygon",
+    "coordinates": [[
+        [7.737228350528245,51.86687168604513],
+        [7.507741544165615,51.86687168604513],
+        [7.507741544165615,52.05013100121914],
+        [7.737228350528245,52.05013100121914],
+        [7.737228350528245,51.86687168604513]
+    ]]
+}
+let datacube = builder.load_collection("TERRASCOPE_S5P_L3_NO2_TD_V1", extent, [`${year}-01-01`, `${year}-12-31`]);
+```
+
+</template>
+<template v-slot:r>
 
+```r
+p = processes()
 
+year = "2019"
+extent = list( # Münster
+  "type" = "Polygon",
+  "coordinates" = list(list(
+    list(5.9521131313239675, 51.78636868072434),
+    list(5.971674074614773, 50.52897921775278),
+    list(9.893643204421315, 50.609735254808925),
+    list(9.864301789485104, 51.870991335721925),
+    list(5.9521131313239675, 51.78636868072434)
+  ))
+)
+datacube = p$load_collection(
+  id = "TERRASCOPE_S5P_L3_NO2_TD_V1",
+  spatial_extent = extent,
+  temporal_extent = list(paste(year, "-01-01", sep = ""), paste(year, "-12-31", sep = ""))
+)
+```
+
+</template>
+</CodeSwitcher>
+
+### 2. Fill gaps
+
+The data cube may contain no-data values due to the removal of clouds in the pre-processing of the collection.
+We'll apply a linear interpolation along the temporal dimension:
+
+<CodeSwitcher :languages="{py: 'Python', js: 'JavaScript', r: 'R'}">
+<template v-slot:py>
+
+```python
+datacube = datacube.apply_dimension(dimension = "t", process = "array_interpolate_linear")
+```
+
+</template>
+<template v-slot:js>
+
+```js
+let interpolate = (data, _, builder) => builder.array_interpolate_linear(data);
+datacube = builder.apply_dimension(datacube, interpolate, "t");
+```
+
+</template>
+<template v-slot:r>
+
+```r
+interpolate = function(data, context) {
+  p$array_interpolate_linear(data = data)
+}
+datacube = p$apply_dimension(data = datacube, dimension = "t", process = interpolate)
+```
+
+</template>
+</CodeSwitcher>
+
+### 3. Smoothen values (optional)
+
+If you want to smoothen the values to get rid of noise for example, we can run a moving average over a certain amount of days over the temporal dimension.
+If you want to work on the raw values, you can also omit this step.
+You can choose the value of the `moving_average_window` variable, but it needs to be at least 3 and an odd integer.
+In the example below we chose 31 for a full month.
```

Comment:
"In the example below we chose 31 for a full month." -> In the example below 31 was chosen to smooth the timeseries with a moving average of a full month.

Also consider to mention that the odd integer specifies the smoothing in number of days.

Reply:
Solved